drivers/lpm/Kconfig

```diff
@@ -0,0 +1,22 @@
+#
+# Copyright (c) 2024 MUNIC SA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# Top-level configuration file for Renesas LPM drivers.
+
+menuconfig LPM
```

Comment:
Since this is a new driver class, this needs to go through architecture review

Reply:
It doesn't even seem like there is a generic API (only lpm_ra2.h) so there is even more work needed there, I guess.

Reply:
> It doesn't even seem like there is a generic API (only lpm_ra2.h) so there is even more work needed there, I guess.

It's not a generic API certainly, but common part for all Renesas RA MCUs. And this IP is critical for almost all peripherals drivers, it's control the power. Current RA4 and RA8 implementations are working only because some peripherals are active by default after POR or MCU's specific LPM activation is hardcoded into the initialization.

Reply:
then the drivers should be using the power management system

Reply:
> then the drivers should be using the power management system

Unfortunately power management system can be disabled. In this case hardware will be disabled too forever (most of pm_* become dummy). It might seem odd to the user.

Reply:
```
config SOC_SERIES_blah
        select PM
```

Reply:
Ok. I'll think about how to implement it better. However, I'm not sure yet if the PM API allows this to be done easily (the LPM needs module id as an argument).